public/src/main/java/com/revenuecat/purchases/models/StoreProduct.kt

```diff
@@ -74,4 +81,6 @@ interface StoreProduct {
         ReplaceWith("id")
     )
     val sku: String
+
+    fun copyWithOfferingId(offeringId: String): StoreProduct
```

Comment:
Hmm this will be public... Not ideal, but not sure how to fix it...

Reply:
I don't think it's a major issue, but we could move it to an abstract superclass for Google and AmazonStoreProduct that implements this.
OfferingParsers would have to downcast do `AbstractStoreProduct` and call it.

```mermaid
classDiagram
    StoreProduct <|-- AbstractStoreProduct
    AbstractStoreProduct <|-- AmazonStoreProduct
    AbstractStoreProduct <|-- GoogleStoreProduct
    <<Interface>> StoreProduct
    <<Abstract>> AbstractStoreProduct
    class StoreProduct{
    }
    class AbstractStoreProduct{
       copyWithOfferingId() AbstractStoreProduct
    }
    class AmazonStoreProduct{
    }
    class GoogleStoreProduct{
    }
```

Reply:
Hmm but `AmazonStoreProduct` and `GoogleStoreProduct` are also public, so they will be accesible from there... A bit more hidden, true, but still accesible. And actually... I believe we would be forced to expose `AbstractStoreProduct` since we need to expose the superclasses of public classes if I'm not wrong.

Loved the diagram btw ❤️

Reply:
THAT^ yes.

maybe on this ticket? https://revenuecats.atlassian.net/browse/CF-1166?search_id=21d271dc-bb1b-4636-9276-242a82986dcf

or @tonidero is there a broader ticket for reconsidering the modules somewhere?

Reply:
I don't think we do. We've been talking about it but haven't created one yet. We've been talking about this project for a bit. CC @aboedo